students/Michael Hrstka/Lab 1/RGB_LED_Design_Challenge/RGB_LED_Design_Challenge.ino

```diff
@@ -0,0 +1,146 @@
+int ledR = 3;
+int ledG = 5;
+int ledB = 6;
+int btn = 12;
+int state = 0;
+int numState = 4;
+
+void setup() {
+  /* Setup pins */
+  pinMode(ledR, OUTPUT);
+  pinMode(ledG, OUTPUT);
+  pinMode(ledB, OUTPUT);
+  pinMode(btn, INPUT);
+}
+
+void chkBtn() {
+  if (digitalRead(btn) == HIGH) {
+    state = state >= numState ? 0 : state + 1;
+    while (digitalRead(btn) == HIGH){ // Debounce! (Wait for release)
+      delay(1);
+    }
+  }
+}
+
+/* Rapid white LED flash */
+bool ledBlink = true;
```

Comment:
for code clarity, try to declare global variables at the top

Reply:
Looking at it now, I definitely could have just had one extra state and one fade variable shared between all of the different methods (instead of policeFadeIn, purpleFadeIn, etc). It would reset to 0 on button press. 

This worked better in my mind at the time to have them all be separate, grouped near each methods to find easier, though obviously was not ideal.